toolkit/tools/internal/customizationmacros/customizationmacros.go

```diff
@@ -0,0 +1,136 @@
+		"This macro file was dynamically generated by the Azure Linux Toolkit image generator",
+		"based on the configuration used at image creation time.",
+		"",
+	}
+	docComments []string = []string{
+		"This stops anything rpm considers a documentation files from being installed.",
+		"To enable documentation files, remove this file, or comment out '%%_excludedocs 1'",
+		"Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+	localeComments []string = []string{
+		"This stops locale files from being installed. %%_install_langs acts as a filter for locales",
+		"which start with the provides strings. Setting it to an invalid value (ie 'NONE') will",
+		"prevent any locale files from being installed.",
+		"To enable locale files, remove this file, or comment out '%%_install_langs <LOCALE STRING>'",
+		"Any packages which are already installed must be reinstalled for this change to take effect.",
+	}
+)
+
+// AddCustomizationMacros adds the currently defined image custimization macros to the specified root directory.
+// For each of disableDocs and disableRpmLocales a macro file is created with the corresponding macros defined in the
+// default rpm macros directory.
+func AddCustomizationMacros(rootDir string, disableDocs, disableRpmLocales bool, overrideRpmLocales string) (err error) {
+	if overrideRpmLocales != "" && disableRpmLocales {
+		return fmt.Errorf("cannot set both DisableRpmLocales and OverrideRpmLocales")
+	}
+
+	macroDir, err := rpm.GetMacroDir()
+	if err != nil {
+		return fmt.Errorf("failed to get rpm macro directory when adding customization macros:\n%w", err)
+	}
+	fullMacroDirPath := filepath.Join(rootDir, macroDir)
+
+	if disableDocs {
+		logger.Log.Debugf("Disabling documentation packages")
+		err = AddMacroFile(fullMacroDirPath, rpm.DisableDocumentationDefines(), disableDocsMacroFile, docComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable docs macro file:\n%w", err)
+		}
+	}
+	if disableRpmLocales {
+		logger.Log.Debugf("Disabling locale packages")
+		err = AddMacroFile(fullMacroDirPath, rpm.DisableLocaleDefines(), configureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add disable locales macro file:\n%w", err)
+		}
+	}
+	if overrideRpmLocales != "" {
+		logger.Log.Debugf("Overriding locale packages with (%s)", overrideRpmLocales)
+		err = AddMacroFile(fullMacroDirPath, rpm.OverrideLocaleDefines(overrideRpmLocales), configureRpmLocalesMacroFile, localeComments)
+		if err != nil {
+			return fmt.Errorf("failed to add override locales macro file:\n%w", err)
+		}
+	}
+	return nil
+}
+
+// formatComments ensures that the comments are valid for a macro file: ie they are empty or start with '#'
+func formatComments(comments []string) (formattedComments []string) {
+	for _, comment := range comments {
+		strippedComment := strings.TrimSpace(comment)
+		if !strings.HasPrefix(strippedComment, "#") && comment != "" {
```

Comment:
Got that feedback elsewhere as well, just felt friendly but I'll simplify. I supose there really isn't any need since no one else will really be using this.